psh/test-cat.py

```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
     psh.assert_cmd(p, 'cat -h', expected=help, result='success')
 
 
+def assert_cat_err(p):
+    fname = 'nonexistentFile'
+
+    statement = f'cat: {fname}: ENOENT'
+    psh.assert_cmd(p, f'cat {fname}', expected=statement, result='fail')
+
+    statement = rf'cat: {ROOT_TEST_DIR}: EISDIR'
+    psh.assert_cmd(p, f'cat {ROOT_TEST_DIR}', expected=statement, result='fail')
+
+
+def assert_cat_basic(p):
+    fname = ROOT_TEST_DIR + "/textFile.txt"
+    secFname = ROOT_TEST_DIR + "/textFile2.txt"
+    fspecial = ROOT_TEST_DIR + "/text!#$?@File.txt"
+
+    text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. "
+    secText = "Curabitur eu velit a sem porttitor blandit."
+    ascii = "0123456789?@ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+
+    assert_file_created(p, fname)
+    assert_file_created(p, secFname)
+    assert_file_created(p, fname)
+
+    # Empty file
+    psh.assert_cmd(p, f'cat {fname}', expected="")
+
+    # Printing content of file
+    psh.assert_cmd(p, f'echo "{text}" > {fname}')
+    psh.assert_cmd(p, f'cat {fname}', expected=text)
+
+    psh.assert_cmd(p, f'echo "{secText}" > {secFname}')
+    psh.assert_cmd(p, f'cat {secFname}', expected=secText)
+
+    # Concrate two text files
+    psh.assert_cmd(p, f'cat {fname} {secFname}', expected=[text, secText])
```

Comment:
Let's use tuple if possible.

```suggestion
    psh.assert_cmd(p, f'cat {fname} {secFname}', expected=(text, secText))
```

Reply:
Applies to the other places.